source/appModules/explorer.py

```diff
@@ -299,3 +300,16 @@ def event_gainFocus(self, obj, nextHandler):
 			return
 
 		nextHandler()
+
+	def isGoodUIAWindow(self,hwnd):
+		# #9204: In build 18305 and later, it is the shell that raises window open event for emoji panel.
+		if winVersion.winVersion.build > 17763 and winUser.getClassName(hwnd) == "ApplicationFrameWindow":
+			return True
+		return False
+
+	def event_UIA_window_windowOpen(self, obj, nextHandler):
+		# Send UIA window open event to input app window.
+		inputPanelWindow = obj.firstChild
+		if isinstance(obj, UIA) and obj.UIAElement.cachedClassName == "ApplicationFrameWindow" and inputPanelWindow and inputPanelWindow.appModule.appName == "windowsinternal_composableshell_experiences_textinput_inputapp":
+			inputPanelWindow.appModule.event_UIA_window_windowOpen(inputPanelWindow, nextHandler)
+		nextHandler()

```

Comment:
You're providing nextHandler to inputPanelWindow.appModule.event_UIA_window_windowOpen and than calling it again here. I'm afraid this can cause weird behaviour. I think I'd prefer using eventHandler.executeEvent on the object, in which case the appModule is triggered automatically.